ui/app/js/controllers/permissions_controller.js

```diff
@@ -66,18 +104,26 @@ function($scope, $routeParams, $location, $timeout, Permissions, Search, $modal,
   $scope.highlightSearch = Search.highlightSearch;
   // $scope.removeFilterSearchWord = Search.removeFilterSearchWord;
 
-  $scope.filterBySearch = function(item) {
+  $scope.filterBySearch = function (item) {
     // basically, look for a reason to NOT include this
     if (Search.word_regexes.length) {
       // every word in the word_regexes array needs to have some match
       var matches = 0;
-      _.each(Search.word_regexes, function(each) {
+      _.each(Search.word_regexes, function (each) {
         var regex = each[0];
         var on = each[1];
         // console.log(regex, on);
-        if ((on === '*' || on === 'username') && item.username && item.username.match(regex)) {
-          matches++;
-          return;
+        if ('username' in item) {
+          if ((on === '*' || on === 'username') && item.username && item.username.match(regex)) {
+            matches++;
+            return;
+          }
+        }
+        else {
```

Comment:
It doesn't look like filterBySearch is enabled for Roles - are you sure you need this section?